server/aesdsocket.c

```diff
@@ -21,57 +21,296 @@
+
+void* thread_rec_data(void* thread_arg)
+{
+    int push_data = 0;
+    //uint32_t total_length = 0;
+    int accepted = 1,file_fd = -1;
+    uint32_t total_length = 0;
+    char rec_val[REC_LEN];
+    memset(rec_val,0,REC_LEN);
+    thread_list *thread_info = (thread_list*) thread_arg;
+    char* store_data = (char*)malloc(sizeof(char)*REC_LEN);
+    if(store_data == NULL)
     {
-        syslog(LOG_INFO,"SIGINT detected");
+        syslog(LOG_ERR,"Unable to allocate memory");
+        printf("malloc failed\n");
+        thread_info->thread_complete_flag = true;
+        return NULL;
     }
-    else if(signo == SIGTERM)
+    else
     {
-        syslog(LOG_INFO,"SIGTERM detected");
+        syslog(LOG_DEBUG,"Successfully_created_file");
+        //thread_info->thread_complete_flag = true;
+        //printf("malloc_succ\n");
     }
-    close(sock_fd);
-    unlink(file_aesdsocket);
-    exit(EXIT_SUCCESS);
+    int recv_len = 0;
+    memset(store_data,0,REC_LEN);
+    //total_length = 0;    
+    while (accepted)
+    {   
+            /* code */
+            int i = 0;
+            recv_len = recv(thread_info->sock_accept_fd,(void*)rec_val,REC_LEN,0);
+            if(recv_len == -1)
+            {
+                perror("recv_error");
+                syslog(LOG_ERR, "Recv ERROR");
+                //printf("recv_error\n");
+                close(thread_info->sock_accept_fd);
+                thread_info->thread_complete_flag = true;
+                //close(thread_info->sock_accept_fd);
+                return NULL;
+            }   
+            else if(recv_len == 0)//Connection failed
+            {
+                //printf("In recvlen 0\n");
+                accepted = 0;
+                total_length = 0;
+                thread_info->thread_complete_flag = true;
+                close(thread_info->sock_accept_fd);
+                break;
+            }           
+            while(i < recv_len)
+            {
+                if(rec_val[i] == '\n')
+                {
+                    push_data = 1;
+                    i++;
+                    break;
+                }
+                i++;    
+            }
+            store_data = (char*)realloc(store_data,total_length+i);
+            memcpy(store_data+total_length,rec_val,i);
+            total_length+=i;
+                
+            memset(rec_val,0,recv_len);
+            if(recv_len !=0 && push_data == 1)
+            {
+                pthread_mutex_lock(&rec_lock);
+                //My data is stored in the buffer, now write it to the file file_aesdsocket
+                file_fd = open(file_aesdsocket,O_WRONLY|O_APPEND);
+                if(file_fd == -1)
+                {
+                    printf("File opening failed\n");
+                    syslog(LOG_ERR, "File opening failed");
+                    close(thread_info->sock_accept_fd);
+                    thread_info->thread_complete_flag = true;
+                    return NULL;
```

Comment:
@SrirajVemparala what happens to `rec_lock` in all of these return early failure paths?